core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java

```diff
@@ -0,0 +1,419 @@
+    MatchPlayer player = PGM.get().getMatchManager().getPlayer(event.getPlayer());
+    if (player != null) {
+      this.render(player);
+    }
+  }
+
+  public void deferredRender() {
+    if (this.deferredRenderTask != null) return;
+
+    // Render one tick later so the effects of the event are visible, and so multiple updates are
+    // batched
+    this.deferredRenderTask =
+        PGM.get()
+            .getServer()
+            .getScheduler()
+            .runTask(
+                PGM.get(),
+                () -> {
+                  Match last = null;
+                  Iterator<Match> it = PGM.get().getMatchManager().getMatches();
+                  while (it.hasNext()) last = it.next();
+                  if (last != null) LegacyMatchTabDisplay.this.render(last);
+                  LegacyMatchTabDisplay.this.deferredRenderTask = null;
+                });
+  }
+
+  public void render(Match match) {
+    int largestTeam = 0;
+    for (Competitor team : match.getCompetitors()) {
+      if (team.getPlayers().size() > largestTeam) {
+        largestTeam = team.getPlayers().size();
+      }
+    }
+
+    // Have a margin between switching modes, so it doesn't flicker back and forth
+    this.compact =
+        match.getCompetitors().size() > WIDTH
+            || largestTeam > MAX_TEAM_SIZE
+            || (this.compact && largestTeam > MAX_TEAM_SIZE - COMPACT_MODE_HYSTERESIS);
+
+    for (MatchPlayer viewer : match.getPlayers()) {
+      this.render(viewer);
+    }
+  }
+
+  public void render(MatchPlayer viewer) {
+    if (viewer.getProtocolVersion() != ViaUtils.VERSION_1_7) return;
+
+    Player bukkit = viewer.getBukkit();
+    MapInfo mapInfo = viewer.getMatch().getMap();
+
+    // Show map name in the top left corner, taking a very rough guess at the length limit.
+    // We could maybe use ChatUtils.pixelWidth() in the future for more accuracy.
+    String mapName = mapInfo.getName();
+    if (mapName.length() > 15) {
+      mapName = mapName.substring(0, 14) + "...";
+    }
+    this.tabDisplay.set(bukkit, 0, 0, ChatColor.AQUA + mapName);
+
+    // If there is exactly one map author, show their name in the top middle slot.
+    // Multiple names will surely not fit, so showing none of them is the only fair thing.
+    if (mapInfo.getAuthors().size() == 1) {
+      this.tabDisplay.set(
+          bukkit,
+          1,
+          0,
+          TextTranslations.translateLegacy(
+              TranslatableComponent.of(
+                  "misc.by",
+                  TextColor.DARK_GRAY,
+                  TextComponent.of(
+                      mapInfo.getAuthors().iterator().next().getNameLegacy(), TextColor.GRAY)),
+              viewer.getBukkit()));
+    } else {
+      this.tabDisplay.set(
+          bukkit,
+          1,
+          0,
+          TextTranslations.translateLegacy(
+              TranslatableComponent.of(
+                  "tablist.authors.tooMany",
```

Comment:
TBH I think we can leave this out. If too many, just show the first one.

Reply:
Showing first author, would be unfair for the other authors on there